### PR TITLE
Add back in heading for P4Smith README file

### PR DESCRIPTION
**Pull request by jafingerhut**
This makes the auto-generated doxygen index easier to distinguish this files from others, instead of merely being called "README" there.

### CODE REVIEW

**Comment by fruffy** [comment]
This will break the formatting of the README itself:
https://github.com/jafingerhut/p4c/tree/add-readme-heading-for-better-doxygen-index/backends/p4tools/modules/smith

image

**Comment by fruffy** [comment]
Looks like the formatting is similarly broken in other places: https://github.com/jafingerhut/p4c/tree/add-readme-heading-for-better-doxygen-index/backends/p4tools/modules/testgen

**Comment by AdarshRawat1** [comment]
> I am open to anything that you find acceptable for the page itself, but it seems very poor to have the Doxygen index of articles include "README" for this file.
> 
> @AdarshRawat1 Any Doxygen configuration tricks you know of that might help here?

# Issue here
The page name and title come from the file name or level one heading, and Standard Markdown does not support header labeling.

# Fix 
We can use PHP Markdown Extra format for labeling our pages 
```
Page Title                         {#label}
============
```

## Output   
image

**Comment by AdarshRawat1** [comment]
> I am open to anything that you find acceptable for the page itself, but it seems very poor to have the Doxygen index of articles include "README" for this file.
> 
> @AdarshRawat1 Any Doxygen configuration tricks you know of that might help here?

# Issue here
The page name and title come from the file name or level one heading, and Standard Markdown does not support header labeling. 

If we can remove the image, then this should sort the issue of header name.






## Possible Alternatives to page naming

We can use `\page` command to give title to our pages. 
```
\page label_Id P4Smith 
```

### Output   
image

### Note 
This leads to the command being displayed in GitHub README output. And [as discussed earlier](https://github.com/p4lang/p4c/issues/4746#issuecomment-2185282972) there is no working for hiding it [Currently]. 

image

**Comment by AdarshRawat1** [comment]
I couldn't find any possible alternative that allows this without distorting our README and Outputs.

**Comment by jafingerhut** [comment]
@fruffy What do you think of Adarsh's idea of removing the P4 logo image, and the second heading, but keeping the new first one?  The test-tools pass/fail badge would stay.

**Comment by fruffy** [comment]
> @fruffy What do you think of Adarsh's idea of removing the P4 logo image, and the second heading, but keeping the new first one? The test-tools pass/fail badge would stay.

Not happy since the logo adds some flavor but it can't be helped. The clean documentation takes priority.

**Comment by jafingerhut** [comment]
Ugh.  Forgot to add DCO to these commits.  Will try to add them retroactively.

**Comment by fruffy** [comment]
DCO is still optional :) I agree it is quite annoying.

**Comment by jafingerhut** [comment]
The instructions for adding DCO retroactively works, even if you forgot them for multiple commits.  Good!

**Comment by fruffy** [comment]
Could also do this for P4Testgen?

**Comment by jafingerhut** [comment]
> Could also do this for P4Testgen?

Done

**Comment by AdarshRawat1** [comment]
> Not happy since the logo adds some flavor but it can't be helped. The clean documentation takes priority.

How about having P4 image on the right-hand side of the page [like this ](https://adarshrawat1.github.io/P4c-Documentation-Hosting/) ??
image

File: backends/p4tools/modules/testgen/README.md [hunk 1]
```diff
@@ -2,8 +2,6 @@
```
Comment (fruffy) [inline backends/p4tools/modules/testgen/README.md:2]: The header is missing here.

Comment (jafingerhut) [reply backends/p4tools/modules/testgen/README.md:2]: The first line is `# P4Testgen`.  I am not sure which header you are referring to.

Comment (fruffy) [reply backends/p4tools/modules/testgen/README.md:2]: Haven't had my coffee yet, never mind :)

**Review by fruffy — APPROVED** [review]

**Pull request merged**